### PR TITLE
Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by paulbalandan**
**Checklist:**
- [x] Securely signed commits

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -95,12 +116,19 @@ The best way to contribute is to fork the CodeIgniter4 repository, and "clone" t
 5. Create a new [branch](https://help.github.com/en/articles/about-branches) in your project for each set of changes you want to make.
 6. Fix existing bugs on the [Issue tracker](https://github.com/codeigniter4/CodeIgniter4/issues) after confirming that no one else is working on them.
 7. [Commit](https://help.github.com/en/desktop/contributing-to-projects/committing-and-reviewing-changes-to-your-project) the changed files in your contribution branch.
-8. [Push](https://docs.github.com/en/github/using-git/pushing-commits-to-a-remote-repository) your contribution branch to your fork.
-9. Send a [pull request](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork).
+8. Commit messages are expected to be descriptive of what you changed specifically. Commit messages like
+"Fixes #1234" would be asked by the reviewer to be revised.
+9. If there are intermediate commits that are not meaningful to the overall PR, it is advised to squash your
```
Comment (MGatner) [inline CONTRIBUTING.md:121]: I'm always a fan of more commits. I think you mean if there are unnecessary commits (like "Revert the mistake I made" or "Try another way fo testing") but maybe we should make that more clear? Or if I misunderstood an you mean squashing in general then maybe we need to have a team conversation about preferred behavior.

**Comment by MGatner** [comment]
Thanks for splitting up the style & content commits! Makes reviews *so* much easier.

**Review by MGatner — CHANGES_REQUESTED** [review]
See comment on 9

File: CONTRIBUTING.md (hunk 1 above)
Comment (paulbalandan) [reply CONTRIBUTING.md:121]: Yes I mean "Reverting a mistake" etc. kind of commits.

**Comment by lonnieezell** [comment]
@paulbalandan Thanks for doing this! And even including some of my minor complaints from last night lol. Looks good.

**Pull request merged**